src/delta.f90

```diff
@@ -644,6 +658,77 @@ subroutine fill_tetrahedra_3d(tetra, evals, tetra_evals)
        end do
     end do
   end subroutine fill_tetrahedra_3d
+  
+  
+subroutine fermi_window_adjusted_fill_tetrahedra_3d(wann, crys, wave_vector_mesh, tetra, evals, tetra_evals)
+
+    type(wannier), intent(in) :: wann
+    type(crystal), intent(in) :: crys
+    integer(i64), intent(in) :: tetra(:,:)
+    real(r64), intent(in) :: evals(:,:)
+    integer(i64), intent(in) :: wave_vector_mesh(1, 3)
+    
+    real(r64), allocatable, intent(out) :: tetra_evals(:,:,:)
+
+    integer(i64) :: iv, it, ib, numbands, aux, numtetra
+    integer(i64) :: kpoint(3)
+    
+    real(r64), allocatable :: energies(:,:)
+    real(r64) :: kvecs(1,3)
```

Comment:
A better name would be `k_frac`, since this is the wave vector expressed in fractional units.